desktop/core/src/desktop/lib/idbroker/conf.py

```diff
@@ -40,32 +40,41 @@ def validate_fs(fs=None):
   if fs in SUPPORTED_FS:
     return SUPPORTED_FS[fs]
   else:
-    LOG.warning('Selected FS %s is not supported by Hue IDBroker client' % fs)
+    LOG.warning('Selected filesystem %s is not supported by Hue IDBroker client.' % fs)
     return None
 
 def _handle_idbroker_ha(fs=None):
+  LOG.debug('47: check fs' + fs)
+
   fs = validate_fs(fs)
-  idbrokeraddr = get_conf().get(_CNF_CAB_ADDRESS % fs) if fs else None
-  response = None
+  LOG.debug('50: check fs again' + fs)
+
+  id_broker_addr_list = []
   if fs:
-    id_broker_addr = get_conf().get(_CNF_CAB_ADDRESS % fs)
-    if id_broker_addr:
-      id_broker_addr_list = id_broker_addr.split(',')
-      for id_broker_addr in id_broker_addr_list:
-        try:
-          response = requests.get(id_broker_addr.rstrip('/') + '/dt/knoxtoken/api/v1/token', auth=HTTPKerberosAuth(), verify=False)
-        except Exception as e:
-          if 'Name or service not known' in str(e):
-            LOG.warn('IDBroker %s is not available for use' % id_broker_addr)
-        # Check response for None and if response code is successful (200) or authentication needed (401)
-        if (response is not None) and (response.status_code in (200, 401)):
-          idbrokeraddr = id_broker_addr
-          break
-      return idbrokeraddr
-    else:
-      return idbrokeraddr
-  else:
-    return idbrokeraddr
+    idbrokeraddr = get_conf().get(_CNF_CAB_ADDRESS % fs, '')
+    LOG.debug('idbroker string from core-site' + idbrokeraddr)
+
+    id_broker_addr_list = idbrokeraddr.split(',')
+
+  response = None
+  for id_broker_addr in id_broker_addr_list:
+    try:
+      response = requests.get(id_broker_addr.rstrip('/') + '/dt/knoxtoken/api/v1/token', auth=HTTPKerberosAuth(), verify=False)
+    except Exception as e:
+      LOG.debug('64: printing idbroker response exception --------------->>>')
+      LOG.debug(str(e))
+
+      if 'Name or service not known' in str(e) or 'Failed to establish a new connection' in str(e):
+        LOG.warn('IDBroker %s is not available for use' % id_broker_addr)
+        LOG.debug('Not connecting to IDbroker check ------------>>>')
+
+    # Check response for None and if response code is successful (200) or authentication needed (401)
+    if (response is not None) and (response.status_code in (200, 401)):
+      idbrokeraddr = id_broker_addr
+      LOG.debug('71: Return idbroker url -------->>>' + idbrokeraddr)
+      break
+
```

Comment:
After iteration and no active ID broker has been found should we not throw an exception "Hue could not complete the task. No active IDBroker instance was found. Please contact your system administrator."

Reply:
_handle_idbroker_ha only returns the correct and healthy IDBroker URL if available else returns None to be used by IDBroker client: https://github.com/cloudera/hue/blob/a17d4995b559e0c0c4dfbb09bbdb48148ad591f1/desktop/core/src/desktop/lib/idbroker/client.py#L32-L47 (line 43 where we send the correct address)

Then when getting the token we catch the exception if call fails and show the message `'Failed to authenticate to IDBroker with error: %s' % e.message`: https://github.com/cloudera/hue/blob/a17d4995b559e0c0c4dfbb09bbdb48148ad591f1/desktop/core/src/desktop/lib/idbroker/client.py#L75-L79

For better error handling we can also add an extra check in the `get_auth_token` method to see if address is `None` and then show `No active IDBroker instance was found. Please contact your system administrator`. 

What do you think, looks good?

Reply:
or return None in that case. But I feel that the overall code won't function without the address so that would warrant an exception.

Reply:
You're right. Fixing it.